packages/lending/src/protocols/spark/lending-protocol.ts

```diff
@@ -0,0 +1,394 @@
+        stableBorrowBalance: common.toBigUnit(currentStableDebt, asset.decimals),
+        variableBorrowBalance: common.toBigUnit(currentVariableDebt, asset.decimals),
+        usageAsCollateralEnabled,
+      };
+    }
+
+    return userBalancesMap;
+  }
+
+  // https://github.com/aave/interface/blob/release-2023-07-25_15-22/src/hooks/app-data-provider/useAppDataProvider.tsx#L228
+  // https://github.com/aave/aave-v3-core/blob/v1.19.1/contracts/protocol/libraries/logic/GenericLogic.sol#L64
+  async getPortfolios(account: string) {
+    const reserveDataMap = await this.getReserveDataMap();
+    const assetPriceMap = await this.getAssetPriceMap();
+    const userBalancesMap = await this.getUserBalancesMap(account);
+
+    const supplies: SupplyObject[] = [];
+    for (const token of tokensForDepositMap[this.chainId]) {
+      if (hasNativeToken(this.chainId) && token.isWrapped) continue;
+
+      const reserveData = reserveDataMap[token.address];
+      const { supplyAPY: apy, ltv, liquidationThreshold, supplyCap, totalSupply, debtCeiling } = reserveData;
+
+      const price = assetPriceMap[token.address];
+
+      const userBalances = userBalancesMap[token.address];
+      const { supplyBalance: balance } = userBalances;
+
+      // https://github.com/aave/interface/blob/release-2023-08-12_03-18/src/components/transactions/utils.ts#L61
+      const usageAsCollateralEnabled = debtCeiling.gt(0)
+        ? false
+        : Number(balance) > 0
+        ? userBalances.usageAsCollateralEnabled
+        : reserveData.usageAsCollateralEnabled;
+
+      supplies.push({
+        token,
+        price,
+        balance,
+        apy,
+        usageAsCollateralEnabled,
+        ltv,
+        liquidationThreshold,
+        supplyCap,
+        totalSupply,
+      });
+    }
+
+    const borrows: BorrowObject[] = [];
+    for (const token of tokensForBorrowMap[this.chainId]) {
+      if (hasNativeToken(this.chainId) && token.isWrapped) continue;
+
+      const { stableBorrowAPY, variableBorrowAPY, borrowCap, totalBorrow } = reserveDataMap[token.address];
+      const price = assetPriceMap[token.address];
+      const { stableBorrowBalance, variableBorrowBalance } = userBalancesMap[token.address];
+
+      borrows.push({
+        token,
+        price,
+        balances: [variableBorrowBalance, stableBorrowBalance],
+        apys: [variableBorrowAPY, stableBorrowAPY],
+        borrowCap,
+        totalBorrow,
+      });
+    }
+
+    const portfolio = new Portfolio(this.chainId, this.id, this.market.id, supplies, borrows);
+
+    return [portfolio];
+  }
+
+  async getPortfolio(account: string) {
+    return this.getPortfolios(account).then((portfolios) => portfolios[0]);
+  }
+
+  override canCollateralSwap() {
+    return true;
+  }
+
+  override canDebtSwap() {
+    return true;
+  }
+
+  override canLeverage(_marketId: string, assetToken: common.Token) {
+    return true;
+  }
+
+  override canDeleverage() {
+    return true;
+  }
```

Comment:
1. Can remove those override functions as default is `true`
2. To support aave-v3 avalanche, we need to update and remove those override functions in `src/protocols/aave-v3/lending-protocol.ts`